utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -199,51 +200,43 @@ impl<'a, 'b> SysCallsInvocator<'a, 'b> {
 
     fn build_sys_call_invoke_instructions(
         &mut self,
-        invocable: InvocableSysCall,
-        call_indexes_handle: CallIndexesHandle,
+        call_data: InvocableSysCallData,
     ) -> Result<Vec<Instruction>> {
+        let invocable = call_data.invocable;
         log::trace!(
             "Random data before building {} sys-call invoke instructions - {}",
             invocable.to_str(),
             self.unstructured.len()
         );
 
-        let insert_error_processing = self
-            .config
-            .error_processing_config()
-            .error_should_be_processed(&invocable);
-        let (fallible, mut signature) = (invocable.is_fallible(), invocable.into_signature());
-
-        if self.is_not_send_sys_call(invocable) {
+        if self.is_sys_call_with_destination(invocable) {
             log::trace!(
-                " -- Generating build call for non-send sys-call {}",
+                " -- Generating build call for {} sys-call with destination",
                 invocable.to_str()
             );
-            return self.build_call(
-                signature,
-                fallible,
-                insert_error_processing,
-                call_indexes_handle,
+
+            self.build_sys_call_with_destination(call_data)
```

Comment:
```suggestion
            self.build_call_with_destination(call_data)
```

maybe like this? As the lower is just `build_call`, not `build_sys_call`.